lib/statsd/instrument/metric_expectation.rb

```diff
@@ -29,7 +29,7 @@ def matches(actual_metric)
         end
       end
 
-      return expected_tags == actual_tags
+      return expected_tags.subset?(actual_tags)
```

Comment:
@wvanbergen, I think this is something you should look at. I'm suggesting that the tags we place in our expectation should at least be a subset of the tags that we see in actual metrics generated by our code .. this is based on the usage I've seen in some of the Shopify tests, whereby developers seem to add assertions and only check for the tags they're interested in. What we had here previously was an exact match check .. which is also correct, just more strict. What would you prefer?